.github/workflows/build.yml

```diff
@@ -0,0 +1,81 @@
+name: build
+on:
+  push:
+#    branches: [ master ] Todo: Uncomment before merging
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+    inputs:
+      release:
+        description: 'Release? yes/no'
+        default: 'no'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          aws-region: eu-west-1
+      - name: Compile
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: wrapper
+          arguments: compileTestKotlin
+      - name: Build
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: build
+        timeout-minutes: 45
+      - name: Upload test reports
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: test-reports
+          path: ./build/reports/tests
+      - name: Upload test results
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: test-results
+          path: ./build/test-results
+  build-check:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - run: echo "All build jobs successful."
+  release:
```

Comment:
The release part is configured in the same manner as it is done i.e. in [jira-actions](https://github.com/atlassian/jira-actions/blob/master/.github/workflows/ci.yml#L46) lib.